pyccel/parser/syntactic.py

```diff
@@ -103,6 +101,23 @@
 from pyccel.parser.base import BasicParser
 from pyccel.parser.base import is_ignored_module
 
+def change_priority( expr ):
+    first  = expr.args[0]
+    second = expr.args[1]
+    if isinstance(second, PyccelOperator):
+        if second.p<=expr.p:
+            a    = first
+            b    = second.args[0]
+            c    = second.args[1]
+            a    = expr.func(a,b)
+            a    = change_priority(a)
+            expr = second.func(a,c)
+        else:
+            second = change_priority(second)
+            expr   = expr.func(first, second)
+
+    return expr
```

Comment:
This function seems to be of the paramount importance. Can you add some explanation in the form of a Python comment, or even better as a docstring?

Reply:
done